unity-renderer/Assets/UIComponents/Scripts/Components/Carousel/CarouselComponentView.cs

```diff
@@ -434,10 +434,12 @@ internal void ResizeAllItems()
 
     internal void DestroyInstantiatedItems()
     {
-        foreach (Transform child in itemsContainer)
+        List<BaseComponentView> itemsToDestroy = ExtractItems();
+        foreach (BaseComponentView itemToDestroy in itemsToDestroy)
         {
-            Destroy(child.gameObject);
+            DestroyImmediate(itemToDestroy.gameObject);
```

Comment:
Why can't we use the regular `Destroy()` ?

Reply:
Ups... good catch! I had to do some tests with the removing of items (using the immediate way) and I forgot to change it back. But yes, it must be a normal `Destroy` instead an immediate one. Fixed! Thanks!